src/Functions/FunctionStringCompare.cpp

```diff
@@ -0,0 +1,446 @@
+             static_cast<FunctionArgumentDescriptor::TypeValidator>(&isUInt),
+             static_cast<FunctionArgumentDescriptor::ColumnValidator>(&FunctionStringCompare::isConstColumn),
+             "UInt"},
+            {"n",
+             static_cast<FunctionArgumentDescriptor::TypeValidator>(&isUInt),
+             static_cast<FunctionArgumentDescriptor::ColumnValidator>(&FunctionStringCompare::isConstColumn),
+             "UInt"},
+        };
+        validateFunctionArguments(*this, arguments, mandatory_args, optional_args);
+
+        return std::make_shared<DataTypeInt8>();
+    }
+
+    DataTypePtr getReturnTypeForDefaultImplementationForDynamic() const override { return std::make_shared<DataTypeInt8>(); }
+
+    ColumnPtr
+    executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & /*result_type*/, size_t /*input_rows_count*/) const override
+    {
+        auto compare_result_column = ColumnInt8::create();
+
+        auto prepare_string_args
+            = [](const ColumnPtr & column) -> std::tuple<const ColumnString *, const ColumnFixedString *, const ColumnConst *>
+        {
+            return {
+                checkAndGetColumn<ColumnString>(column.get()),
+                checkAndGetColumn<ColumnFixedString>(column.get()),
+                checkAndGetColumn<ColumnConst>(column.get())};
+        };
+        auto [str1_col, fixed_str1_col, const_str1_col] = prepare_string_args(arguments[0].column);
+        auto [str2_col, fixed_str2_col, const_str2_col] = prepare_string_args(arguments[1].column);
+
+        auto str1_offset = arguments.size() > 2 ? arguments[2].column->getUInt(0) : 0;
+        auto str2_offset = arguments.size() > 2 ? arguments[3].column->getUInt(0) : 0;
+        auto bytes = arguments.size() > 2 ? arguments[4].column->getUInt(0) : std::numeric_limits<size_t>::max();
+        if (!bytes)
+        {
+            compare_result_column->insertMany(0, arguments[0].column->size());
+            return std::move(compare_result_column);
+        }
+
+        compare_result_column->getData().resize(arguments[0].column->size());
+        auto & result_array = compare_result_column->getData();
+
+        if (str1_col && const_str2_col)
+        {
+            executeStringConst<false>(*str1_col, str1_offset, String(const_str2_col->getDataAt(0)), str2_offset, bytes, result_array);
+        }
+        else if (const_str1_col && str2_col)
+        {
+            executeStringConst<true>(*str2_col, str2_offset, String(const_str1_col->getDataAt(0)), str1_offset, bytes, result_array);
+        }
+        else if (str1_col && str2_col)
+        {
+            executeStringString(*str1_col, str1_offset, *str2_col, str2_offset, bytes, result_array);
+        }
+        else if (fixed_str1_col && fixed_str2_col)
+        {
+            executeFixedStringFixedString(*fixed_str1_col, str1_offset, *fixed_str2_col, str2_offset, bytes, result_array);
+        }
+        else if (fixed_str1_col && str2_col)
```

Comment:
Can we please remove support for comparisons between `String` and `FixedString`. They seem obscure and if users want to do that, they can easily cast one of the inputs.

Reply:
`StringComparisonImpl` supports comparisons between `String` and `FixedString`.